torch/_inductor/ir.py

```diff
@@ -2694,18 +2694,18 @@ def codegen_unbacked_symbol_defs(self, wrapper):
         for i, s in enumerate(self.get_size()):
             if s in symbols_to_define:
                 wrapper.writeline(
-                    f"{wrapper.declare}{s} = {self.get_name()}.size({i}){wrapper.ending}"
+                    f"{wrapper.codegen_unbacked_symbol_decl(s)} = {self.get_name()}.size({i}){wrapper.ending}"
```

Comment:
A naive question - where do the duplicate unbacked symbols come from in the first place? I am not familiar with how the unbacked symbol defs are created, so I am wondering if it is legitimate to have such dups?

Reply:
Good question. In the test example, it looks like the second `index.Tensor` should be CSE-ed. I think unbacked symbols should be unique in general (can't reuse symbols just for any random expressions). @ezyang would be the best person to answer this question.

```
                index_1 = torch.ops.aten.index.Tensor(x, [eq])
                sin = torch.ops.aten.sin.default(index_1)
                index_2 = torch.ops.aten.index.Tensor(x, [eq])
```